packages/synthesizer-ui/package.json

```diff
@@ -1,16 +1,22 @@
 {
   "name": "@penrose/synthesizer-ui",
   "private": true,
-  "version": "0.0.0",
+  "version": "1.3.0",
   "scripts": {
     "dev": "vite",
     "watch": "vite",
-    "clean": "rimraf dist",
     "start": "vite",
     "typecheck": "tsc --noEmit",
     "build": "NODE_OPTIONS='--max-old-space-size=8192' vite build",
     "preview": "vite preview"
   },
+  "turbo": {
+    "build": "out: [dist/**]",
+    "dev": "cache: false, deps: [^build]",
```

Comment:
Another thing: `build` is defined per repo and referenced as `build` locally. How does turbo find out which `build` to execute? I guess if I do `yarn build --filter=editor`, `turbo` knows it will need to execute `build` for both `core` and `examples` because `package.json` states the dependencies?

Also not sure about the semantics of `^build` in a package vs. in a workspace `package.json`

Reply:
My intuition is:

- `cache` tasks that are basically deterministic processes that take input on disk, produce output on disk (always in the same place), and terminate.
- Don't `cache` tasks that have unclear semantics (e.g. could do a variety of different things), or take in user input, or don't terminate (e.g. `watch`).

For `build` and `^build`: our Turborepo config defines a `build` task for _all_ packages. As you can see in this file, we define the YAML config for the `build` task in `@penrose/synthesizer-ui` to be `{out: [dist/**]}`, which our `turbo.js` script turns into `"@penrose/synthesizer-ui#build": { "dependsOn": ["^build"], "outputs": ["dist/**"] }`, which overrides the default `build` config for that particular instance of the `build` task.

When you do `yarn build --filter=editor`, Turborepo similarly sees that our `turbo.json` config says `"@penrose/editor#build": { "dependsOn": ["^build"], "outputs": ["dist/**", "public/**"] }`, so it looks at the `build` tasks for `@penrose/components` and `@penrose/core`, and builds out the graph of all their dependencies, then runs all those tasks (some of which are `build` tasks, some of which are not).

Reply:
Makes sense. Pretty much my understanding of the semantics too. Thanks for explaining!